esmvaltool/diag_scripts/ocean/diagnostic_maps_multimodel.py

```diff
@@ -0,0 +1,316 @@
+    model difference (or data) organized in rows/cols using row/col layout.
+
+    Parameters
+    ----------
+    cfg: dict
+        the opened global config dictionary, passed by ESMValTool.
+    metadata: dict
+        the input files dictionairy
+    obs_filename: str
+        the preprocessed observations file.
+    """
+    logger.debug('make_multiple_plots')
+    # ####
+    filenames = list(metadata.keys())
+    proj = ccrs.Robinson(central_longitude=0)
+    # plot layout
+    layout = metadata[filenames[0]]['layout_rowcol']
+
+    # check if observations are provided
+    hasobs = False
+    obsname = ''
+    if obs_filename != '':
+        hasobs = True
+        obsname = metadata[obs_filename]['dataset']
+        filenames.remove(obs_filename)
+        filenames.insert(0, obs_filename)
+        layout[0] = layout[0] + 1
+    else:
+        logger.info('Observations not provided. Plot each model data.')
+
+    # Load the data for each layer as a separate cube
+    layers = {}
+    cubes = {}
+    for thename in filenames:
+        logger.debug('loading: \t%s', thename)
+        cube = iris.load_cube(thename)
+        cube = diagtools.bgc_units(cube, metadata[thename]['short_name'])
+        model_name = metadata[thename]['dataset']
+        cubes[model_name] = diagtools.make_cube_layer_dict(cube)
+        for layer in cubes[model_name]:
+            layers[layer] = True
+
+    logger.debug('layers: %s', layers)
+    logger.debug('cubes: %s', ', '.join(cubes.keys()))
+
+    # Load image format extention
+    image_extention = diagtools.get_image_format(cfg)
+
+    # Make a plot for each layer
+    extend = 'neither'
+    for layer in layers:
+
+        fig = plt.figure()
+        fig.set_size_inches(layout[1] * 4., layout[0] * 2. + 2.)
+
+        # aggregate cubes
+        name_cubes = []
+        maps_cubes = []
+        diff_cubes = []
+        for thename in cubes:
+            maps_cubes.append(cubes[thename][layer])
+            name_cubes.append(thename)
+            if (hasobs) & (thename != obsname):
+                diff_cubes.append(cubes[thename][layer] -
+                                  cubes[obsname][layer])
+
+        # data ranges
+        maps_range = diagtools.get_cube_range(maps_cubes)
+        if hasobs:
+            diff_range = diagtools.get_cube_range(diff_cubes)
+            maps_range = diagtools.get_cube_range([cubes[obsname][layer]])
+
+        if 'maps_range' in metadata[filenames[0]]:
+            maps_range = metadata[filenames[0]]['maps_range']
+            extend = 'both'
+        if 'diff_range' in metadata[filenames[0]]:
+            diff_range = metadata[filenames[0]]['diff_range']
+            extend = 'both'
+
+        # create subplots
+
+        varunit = str(maps_cubes[0].units)
+        varname = maps_cubes[0].var_name
+        gsc = gridspec.GridSpec(layout[0], layout[1])
+        yy = 0
+        xx = 0
+        clevels = 13
+        for ii in range(len(maps_cubes)):
```

Comment:
pls use `for ii, _ in enumerate(maps_cubes):`

Reply:
ok